docs/ClangFormat.md

```diff
@@ -1,21 +1,84 @@
-# Overview
+# Using clang-format in HSE
```

Comment:
Seems unnecessary to carry this file forward. Let's remove it and consolidate it into the CONTRIBUTING file.

In addition. Running clang-format across the files we want should be a ninja target and a shell script.

Reply:
I'll close this PR, you can update CONTRIBUTING.md if you feel inclined.